src/web_peer_connection.cpp

```diff
@@ -789,9 +789,11 @@ void web_peer_connection::on_receive(error_code const& error
 			// if the status code is not one of the accepted ones, abort
 			if (!is_ok_status(m_parser.status_code()))
 			{
-				file_request_t const& file_req = m_file_requests.front();
-				m_web->have_files.resize(t->torrent_file().num_files(), true);
-				m_web->have_files.clear_bit(file_req.file_index);
+				if (!m_file_requests.empty())
+				{
+					file_request_t const& file_req = m_file_requests.front();
+					m_web->have_files.clear_bit(file_req.file_index);
```

Comment:
in next PR can add logic to exclude file idx bit only for (>=400 && < 500) ?  HTTP Status 4XX shows that resource can not be reached by current request